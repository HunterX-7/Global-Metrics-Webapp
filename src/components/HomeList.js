import PropTypes, { shape } from 'prop-types';
import DetailCard from './DetailCard';

const HomeList = (props) => {
  const { data } = props;

  return (
    <div>
      {data.map((country) => {
        const {
          id, name, population, flag,
        } = country;

        return (
          <DetailCard
            key={id}
            name={name}
            population={population}
            flag={flag}
          />
        );
      })}
    </div>
  );
};

export default HomeList;

HomeList.propTypes = {
  data: PropTypes.arrayOf(shape).isRequired,
};
