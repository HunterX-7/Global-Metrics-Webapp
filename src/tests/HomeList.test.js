import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import HomeList from '../components/HomeList';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';

describe('Testing page deployment', () => {
  const data1 = [
    {
      name: 'MockCountry1',
      capital: 'MockCapital1',
      region: 'MockRegion1',
      subregion: 'MockSubregion1',
      area: 'MockArea1',
      maps: 'www.google.com/maps/mockcountry1',
      timezones: 'UTC1',
      continents: 'Continent1',
      population: 1000000001,
      flag: 'www.mockflag.com/1',
      flagDescription: 'Mock flag description 1',
      coatOfArms: 'www.mockcoatofarms.com/1',
      id: 12341,
    },
  ];

  test('HomeList test 1 component matches snapshot', () => {
    const tree = render(
      <HashRouter>
        <Provider store={store}>
          <HomeList data={data1} />
        </Provider>
      </HashRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  const data2 = [
    {
      name: 'MockCountry2',
      capital: 'MockCapital2',
      region: 'MockRegion2',
      subregion: 'MockSubregion2',
      area: 'MockArea2',
      maps: 'www.google.com/maps/mockcountry2',
      timezones: 'UTC2',
      continents: 'Continent2',
      population: 1000000002,
      flag: 'www.mockflag.com/2',
      flagDescription: 'Mock flag description 2',
      coatOfArms: 'www.mockcoatofarms.com/2',
      id: 12342,
    },
  ];

  test('HomeList test 2 component matches snapshot', () => {
    const tree = render(
      <HashRouter>
        <Provider store={store}>
          <HomeList data={data2} />
        </Provider>
      </HashRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
