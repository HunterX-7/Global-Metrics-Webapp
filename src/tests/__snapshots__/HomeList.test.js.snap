// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing page deployment HomeList test 1 component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-3 d-flex justify-content-center col-xl-3 col-lg-4 col-md-6 col-12"
      >
        <div
          class="shadow-lg bg-color2 card"
          style="width: 18rem;"
        >
          <a
            href="#/country/MockCountry1"
          >
            <img
              alt="MockCountry1"
              class="card-img-top p-2"
              src="www.mockflag.com/1"
            />
          </a>
          <div
            class="d-flex flex-column justify-content-evenly card-body"
          >
            <div
              class="text-center card-title h5"
              style="font-weight: 600; font-size: 24px;"
            >
              MockCountry1
            </div>
            <div
              class="text-center card-subtitle h6"
              style="font-weight: 500;"
            >
              MockCapital1
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-3 d-flex justify-content-center col-xl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="shadow-lg bg-color2 card"
        style="width: 18rem;"
      >
        <a
          href="#/country/MockCountry1"
        >
          <img
            alt="MockCountry1"
            class="card-img-top p-2"
            src="www.mockflag.com/1"
          />
        </a>
        <div
          class="d-flex flex-column justify-content-evenly card-body"
        >
          <div
            class="text-center card-title h5"
            style="font-weight: 600; font-size: 24px;"
          >
            MockCountry1
          </div>
          <div
            class="text-center card-subtitle h6"
            style="font-weight: 500;"
          >
            MockCapital1
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing page deployment HomeList test 2 component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-3 d-flex justify-content-center col-xl-3 col-lg-4 col-md-6 col-12"
      >
        <div
          class="shadow-lg bg-color2 card"
          style="width: 18rem;"
        >
          <a
            href="#/country/MockCountry2"
          >
            <img
              alt="MockCountry2"
              class="card-img-top p-2"
              src="www.mockflag.com/2"
            />
          </a>
          <div
            class="d-flex flex-column justify-content-evenly card-body"
          >
            <div
              class="text-center card-title h5"
              style="font-weight: 600; font-size: 24px;"
            >
              MockCountry2
            </div>
            <div
              class="text-center card-subtitle h6"
              style="font-weight: 500;"
            >
              MockCapital2
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-3 d-flex justify-content-center col-xl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="shadow-lg bg-color2 card"
        style="width: 18rem;"
      >
        <a
          href="#/country/MockCountry2"
        >
          <img
            alt="MockCountry2"
            class="card-img-top p-2"
            src="www.mockflag.com/2"
          />
        </a>
        <div
          class="d-flex flex-column justify-content-evenly card-body"
        >
          <div
            class="text-center card-title h5"
            style="font-weight: 600; font-size: 24px;"
          >
            MockCountry2
          </div>
          <div
            class="text-center card-subtitle h6"
            style="font-weight: 500;"
          >
            MockCapital2
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
